docs/proposals/hpa/cronfederatedhpa.md

```diff
@@ -0,0 +1,393 @@
+	// +required
+	Schedule string `json:"schedule"`
+
+	// TargetReplicas is the target replicas to be scaled for deployment, statefulset, or
+	// other resources that have scale sub-resource.
+	// It should be set if ScaleTargetRef is not FederatedHPA.
+	// It shouldn't be set if ScaleTargetRef is FederatedHPA.
+	// +optional
+	TargetReplicas *int32 `json:"targetReplicas,omitempty"`
+
+	// TargetMinReplicas is the target MinReplicas to be set for FederatedHPA.
+	// If you not set this field, the value will not be updated.
+	// It shouldn't be set if ScaleTargetRef is not FederatedHPA.
+	// It should be set if ScaleTargetRef is FederatedHPA.
+	// +optional
+	TargetMinReplicas *int32 `json:"targetMinReplicas,omitempty"`
+
+	// TargetMaxReplicas is the target MaxReplicas to be set for FederatedHPA.
+	// If you not set this field, the value will not be updated.
+	// It shouldn't be set if ScaleTargetRef is not FederatedHPA.
+	// It should be set if ScaleTargetRef is FederatedHPA.
+	// +optional
+	TargetMaxReplicas *int32 `json:"targetMaxReplicas,omitempty"`
+
+	// Suspend is used to suspend the CronFederatedHPARule.
+	// The suspended rule will not be executed.
+	// Default is false
+	// +optional
+	Suspend *bool `json:"suspend,omitempty"`
+
+	// TimeZone for the giving rule, if not specified, the system time zone will be used.
+	// the system time zone comes from the running os of karmada-controller-mamanger component.
+	// Invalid TimeZone will be rejected when applying by karmada-webhook.
+	// see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones for the all timezones.
+	// +optional
+	TimeZone *string `json:"timeZone,omitempty"`
+}
+
+type ScaleTargetRef struct {
+	// APIVersion of the referent resources.
+	// +required
+	ApiVersion string `json:"apiVersion"`
+
+	// Kind of the referent resources.
+	// +required
+	Kind string `json:"kind"`
+
+	// Name of the referent resources.
+	// +required
+	Name string `json:"name"`
+}
+
+type CronFederatedHPAStatus struct {
+	// History is the execution history of the CronFederatedHPA rules.
+	History []RuleExecutionHistory `json:"history"`
+}
+
+type RuleExecutionHistory struct {
+	// RuleName is the name of the CronFederatedHPARule.
+	RuleName string `json:"ruleName"`
+	// NextExecutionTime is the next time to execute the CronFederatedHPARule.
+	NextExecutionTime metav1.Time `json:"nextExecutionTime"`
+	// State is the current state of the CronFederatedHPARule.
+	// Success - this rule is successfully applied.
+	// Failure - this rule is failed to apply.
+	// Skip - this rule is skipped.
+	// Pending - this rule is waiting for execution.
+	ExecutionState ExecutionState `json:"executionState"`
+	// LastExecutionTime is the last time we execute the CronFederatedHPARule.
+	LastExecutionTime metav1.Time `json:"lastExecutionTime"`
+	// Message is a human readable message indicating details about the rule.
+	Message string `json:"message"`
```

Comment:
Is this `Message` field required?

Reply:
https://github.com/karmada-io/karmada/pull/3664#discussion_r1235163676 Check here, it's used to give some hints about the execution result.